app/src/main/java/com/github/houseorganizer/houseorganizer/location/LocationHelpers.java

```diff
@@ -0,0 +1,98 @@
+package com.github.houseorganizer.houseorganizer.location;
+
+import android.Manifest;
+import android.app.Activity;
+import android.app.AlertDialog;
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.location.Location;
+
+import androidx.core.app.ActivityCompat;
+import androidx.core.content.ContextCompat;
+
+import com.github.houseorganizer.houseorganizer.R;
+import com.google.firebase.firestore.DocumentSnapshot;
+import com.google.firebase.firestore.QuerySnapshot;
+
+public class LocationHelpers {
+
+    public static final int PERMISSION_FINE_LOCATION = 99;
+
+    public static DocumentSnapshot getClosestHouse(QuerySnapshot households, Location location){
+        // Get current coordinates
+        double lon = location.getLongitude();
+        double lat = location.getLongitude();
+        double closest = Double.MAX_VALUE;
+        DocumentSnapshot closestHouse = null;
+        for(DocumentSnapshot house : households){
+            double distance = calculateDistance(lon, lat,
+                    house.getDouble("longitude"), house.getDouble("latitude"));
+            if(distance < closest){
+                closest = distance;
+                closestHouse = house;
+            }
+        }
+        return closestHouse;
+    }
+
+    public static double calculateDistance(double lon1, double lat1,
+                                        double lon2, double lat2){
+
+        lon1 = Math.toRadians(lon1);
+        lon2 = Math.toRadians(lon2);
+        lat1 = Math.toRadians(lat1);
+        lat2 = Math.toRadians(lat2);
+
+        // Haversine formula
+        double dlon = lon2 - lon1;
+        double dlat = lat2 - lat1;
+        double temp = Math.pow(Math.sin(dlat / 2), 2)
+                + Math.cos(lat1) * Math.cos(lat2)
+                * Math.pow(Math.sin(dlon / 2),2);
+
+        temp = 2 * Math.asin(Math.sqrt(temp));
+
+        // Radius of earth in kilometers
+        double r = 6371;
+
+        // calculate the result
+        return(temp * r);
```

Comment:
since we only care about relative distance, you might be able to remove the multiplication by r

Reply:
It could be omitted but as the maths functions are approximate I am afraid the difference will be in the error margin without the mutliplication
What do you think ?

Reply:
thats also true, you can leave it like that since it was mostly me being nitpicky to start with